### PR TITLE
refactor(mobile): alternate approach to reading wallets, accounts

### PR DESCRIPTION
**Pull request by kyranjamie**
This PR renames some state to be more consistent with the extension, and introduces the loader pattern handle conditionality of potentially-undefined data that the source.

### CODE REVIEW

File: apps/mobile/src/app/wallet/wallet-settings/configure/[wallet]/[account].tsx [hunk 1]
```diff
@@ -5,34 +5,29 @@ import { useSafeAreaInsets } from 'react-native-safe-area-context';
...
   account: Account;
-}) {
+}
+function ConfigureAccount({ fingerprint, accountIndex, account }: ConfigureAccountProps) {
   const { bottom } = useSafeAreaInsets();
   const theme = useTheme<Theme>();
   const accountNameModalRef = useRef<BottomSheetModal>(null);
   const dispatch = useAppDispatch();
   const navigation = useNavigation();
+
```
Comment (pete-watters) [inline apps/mobile/src/app/wallet/wallet-settings/configure/[wallet]/[account].tsx:30]: I know it's not your code but often on mobile PRs now I see lint issues flagged like:
Screenshot 2024-09-06 at 09 19 33
Screenshot 2024-09-06 at 09 19 43

Maybe we can make a team effort to fix them in PRs as we see them?

File: apps/mobile/src/app/wallet/wallet-settings/configure/[wallet]/index.tsx [hunk 2]
```diff
@@ -35,7 +33,10 @@ import {
   TrashIcon,
 } from '@leather.io/ui/native';
 
-function ConfigureWallet({ fingerprint, wallet }: { fingerprint: string; wallet: WalletStore }) {
+interface ConfigureWalletProps {
+  wallet: WalletStore;
+}
+function ConfigureWallet({ wallet }: ConfigureWalletProps) {
```
Comment (pete-watters) [inline apps/mobile/src/app/wallet/wallet-settings/configure/[wallet]/index.tsx:39]: I'd love a double destructure here 🤤 

`{ wallet: { name, fingerprint .. }  }`

File: apps/mobile/src/locales/index.ts [hunk 3]
```diff
@@ -25,3 +27,11 @@ export const LOCALES = [
 export const LOCALE_CODES = LOCALES.map(({ locale }) => locale);
 
 export const DEFAULT_LOCALE = 'en';
+
+export function initiateI18n() {
```
Comment (pete-watters) [inline apps/mobile/src/locales/index.ts:31]: Can you please educate me on this? Is it to better load more locales?

**Review by pete-watters — APPROVED** [review]
Nice work 👍 I added some feedback but LGTM. 

Good refactor

File: apps/mobile/src/locales/index.ts (hunk 3 above)
Comment (kyranjamie) [reply apps/mobile/src/locales/index.ts:31]: This was just copied from another folder, we had i18n/ and locales/ before. Figure one is enough to tidy root

**Comment by kyranjamie** [comment]
@fbwoolf would love your input here as well, but merging because there are lots of file renames and want to avoid conflicts 😬

**Pull request merged**